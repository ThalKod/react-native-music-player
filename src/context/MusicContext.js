import React from "react";

const MusicContext = React.createContext({
  selected: {},
  list: [],
  changeSelected: () => {}
});

export const MusicProvider = MusicContext.Provider;
export const MusicConsumer = MusicContext.Consumer;

export default MusicContext;
