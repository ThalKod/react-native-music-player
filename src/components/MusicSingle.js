import React, { useContext } from "react";
import {
  View,
    Text,
    StyleSheet,
    Image,
    TouchableOpacity
} from "react-native";
import Proptypes from "prop-types";
import defaultImg from "../../assets/images/default_music_cover.png";
import MusicContext from "../context/MusicContext";
import { moderateScale } from "react-native-size-matters";

const MusicSingle = ({ title, artist, imageUrl, fileName, id }) => {
  const { changeSelected, list } = useContext(MusicContext);

  const onMusicPress = () => {
    const selected = list.find(elm => elm.id === id );
    changeSelected(selected);
  };

  return (
    <TouchableOpacity style={styles.container} onPress={onMusicPress}>
      <View style={styles.imageContainer}>
        <Image
            source={ imageUrl ? {uri: imageUrl} : defaultImg}
            style={styles.image}
        />
      </View>
      <View style={styles.descriptionContainer}>
        <Text style={styles.textStyle}>{title || fileName.substring(0, 25) + "..."}</Text>
        <Text style={[styles.textStyle, { fontSize: 16, fontFamily: "Avenir-Roman" }]}>{artist || "Unknown Artist" }</Text>
      </View>
    </TouchableOpacity>
  )
};

MusicSingle.propTypes = {
  title: Proptypes.string,
  artist: Proptypes.string,
  imageUrl: Proptypes.string,
  filename: Proptypes.string,
  id: Proptypes.string,
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    marginTop: moderateScale(15),
    height: moderateScale(75),
    justifyContent: "space-between",
  },
  imageContainer: {
    height: "100%",
    width: "25%",
    marginRight: moderateScale(20)
  },
  image: {
    borderRadius: moderateScale(10),
    flex: 1,
    width: null,
    height: null,
    resizeMode: 'cover',
  },
  descriptionContainer: {
    flex: 2,
    justifyContent: "space-around",
  },
  textStyle: {
    fontFamily: "Avenir-Black",
    color: "#353840",
    fontSize: moderateScale(18)
  }
});

export default MusicSingle;
