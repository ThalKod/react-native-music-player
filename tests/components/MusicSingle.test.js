import React from "react";
import { create } from 'react-test-renderer';
import { TouchableOpacity } from "react-native";
import MusicSingle  from "../../src/components/MusicSingle";

import { musicData } from "../__mock__/index";
import MusicContext from "../../src/context/MusicContext";

describe("Component MusicSingle", () => {
  test("Should correctly render the component", () => {
    const tree = create(<MusicSingle {...musicData[0]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should change the selected music.", () => {
    const changeSelectedMock = jest.fn();
    const context = {
      list: [],
      changeSelected: changeSelectedMock
    };

    const element = create(
        <MusicContext.Provider value={context} >
          <MusicSingle {...musicData[0]}/>
        </MusicContext.Provider>
    );

    const instance = element.root;
    const root =  instance.findByType(TouchableOpacity);
    root.props.onPress();
    expect(changeSelectedMock).toBeCalled();
  })
});
