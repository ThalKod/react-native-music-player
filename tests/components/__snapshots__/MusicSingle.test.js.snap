// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MusicSingle Should correctly render the component 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "height": 117.85714285714286,
      "justifyContent": "space-between",
      "marginTop": 23.57142857142857,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
        "marginRight": 31.428571428571427,
        "width": "25%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "imageurl",
        }
      }
      style={
        Object {
          "borderRadius": 15.714285714285714,
          "flex": 1,
          "height": null,
          "resizeMode": "cover",
          "width": null,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 2,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#353840",
          "fontFamily": "Avenir-Black",
          "fontSize": 28.285714285714285,
        }
      }
    >
      Test music title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#353840",
            "fontFamily": "Avenir-Black",
            "fontSize": 28.285714285714285,
          },
          Object {
            "fontFamily": "Avenir-Roman",
            "fontSize": 16,
          },
        ]
      }
    >
      Artist
    </Text>
  </View>
</View>
`;
